hadoop-ozone/dist/src/main/smoketest/upgrade/validate.robot

```diff
@@ -28,3 +29,25 @@ Read data from previously created key
     ${random} =         Generate Random String  5  [NUMBERS]
     ${output} =         Execute          ozone sh key get /${PREFIX}-volume/${PREFIX}-bucket/${PREFIX}-key /tmp/key-${random}
                         Should not contain  ${output}       Failed
+
+Setup credentials for S3
+# TODO: Run "Setup secure v4 headers" instead when security is enabled
+    Run Keyword         Setup dummy credentials for S3
+
+Read data from previously created key using S3 API
+    ${result} =         Execute AWSS3APICli and checkrc    get-object --bucket ${PREFIX}-bucket --key key1 /tmp/key1.result    0
+    # Note: "Compare files" doesn't work on NOTICE.txt as it is updated in new Ozone versions.
+#                        Compare files    /opt/hadoop/NOTICE.txt    /tmp/key1.result
+                        Execute and checkrc    rm /tmp/key1.result    0
+
+Create bucket using S3 API if the bucket doesn't exist
```

Comment:
Can we move the aws client writes to generate.robot and leave the aws client reads in validate.robot? It seems a little confusing now having validate.robot do generate and validate for S3.